mlflow/models/evaluation/default_evaluator.py

```diff
@@ -1444,7 +1444,7 @@ def predict_with_latency(X_copy):
                 i, row_data = row
                 single_input = row_data.to_frame().T if is_dataframe else row_data
                 start_time = time.time()
-                y_pred = self.model_predict_fn(single_input)
+                y_pred = self.predict_fn(single_input)
```

Comment:
It doesn't.

Reply:
I checked this line https://github.com/mlflow/mlflow/blob/68bd3ec5378aa2b5f0548fdfddaad28fe2d13c3c/mlflow/models/evaluation/default_evaluator.py#L1895

```
            self.predict_fn, self.predict_proba_fn = _extract_predict_fn(
                model, self.raw_model, self.model_predict_fn
            )
```
the model should be baseline_model in your case, but maybe `self.raw_model` is wrongly set to candidate model.

Reply:
@WeichenXu123 

`_extract_predict_fn` returns `self.model_predict_fn` and `self.model_predict_fn` returns `model.predict`

https://github.com/mlflow/mlflow/blob/2e065e74de78bd588e8d5e4a3c9807dd05545760/mlflow/models/evaluation/default_evaluator.py#L1970

Reply:
Yes https://github.com/mlflow/mlflow/pull/12215#discussion_r1624131527 this should be a buggy place.
according to the logic here, `_extract_predict_fn` should extract predict_fn from the `model` object, i.e. if `model` is candidate model, it should return candidate model predict fn, otherwise baseline model predict fn